ext/sockets/sockets.c

```diff
@@ -1911,15 +1940,23 @@ PHP_FUNCTION(socket_set_option)
 				RETURN_THROWS();
 			}
 
-			convert_to_long(sec);
-			convert_to_long(usec);
+			zend_long zsec = zval_try_get_long(sec, &failed);
```

Comment:
Naming is confusing here. `zsec` would appear to be a `zval`, and `sec` its corresponding `long`, but it's the other way round.